airbyte-webapp/src/components/ArrayOfObjectsEditor/components/EditorHeader.test.tsx

```diff
@@ -0,0 +1,50 @@
+import { render } from "utils/testutils";
+
+import { EditorHeader } from "./EditorHeader";
+
+describe("<ArrayOfObjectsEditor />", () => {
+  let container: HTMLElement;
+  describe("edit mode", () => {
+    beforeEach(async () => {
+      const renderResult = await render(
+        <EditorHeader
+          mainTitle={<div data-testid="mainTitle">"This is the main title"</div>}
+          addButtonText={<div data-testid="addButtonText">"button text"</div>}
+          itemsCount={0}
+          onAddItem={() => {
+            return null;
+          }}
+          mode="edit"
+        />
+      );
+      container = renderResult.container;
+    });
+    test("it renders only relevant items for the mode", () => {
+      const mainTitle = container.querySelector("div[data-testid='mainTitle']");
+      const addButtonText = container.querySelector("div[data-testid='addButtonText']");
+      expect(mainTitle).toBeInTheDocument();
+      expect(addButtonText).toBeInTheDocument();
+    });
+  });
+  describe("readonly mode", () => {
+    beforeEach(async () => {
+      const renderResult = await render(
+        <EditorHeader
+          mainTitle={<div data-testid="mainTitle">"This is the main title"</div>}
+          addButtonText={<div data-testid="addButtonText">"button text"</div>}
+          itemsCount={0}
+          onAddItem={() => {
+            return null;
+          }}
+          mode="readonly"
+        />
+      );
+      container = renderResult.container;
+    });
+    test("it renders only relevant items for the mode", () => {
+      const mainTitle = container.querySelector("div[data-testid='mainTitle']");
+      expect(mainTitle).toBeInTheDocument();
+      expect(container.querySelector("div[data-testid='addButtonText']")).not.toBeInTheDocument();
+    });
+  });
```

Comment:
Nitpick: The setup of rendering things in the beforeEach and just checking a top level variable is arguably a little misleading. Simply rendering what you want in each test would be just as effective, without giving the notion that the values are used elsewhere.